extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ChunkReaderFactory.java

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.extensions.barrage.chunk;
+
+import io.deephaven.chunk.ChunkType;
+import io.deephaven.extensions.barrage.util.StreamReaderOptions;
+import org.apache.arrow.flatbuf.Field;
+import org.apache.arrow.flatbuf.Type;
+
+/**
+ * Supports creation of {@link ChunkReader} instances to use when processing a flight stream. JVM implementations for
+ * client and server should probably use {@link DefaultChunkReadingFactory#INSTANCE}.
+ */
+public interface ChunkReaderFactory {
```

Comment:
Thoughts on this being an inner interface to ChunkReader, perhaps as ChunkReader.Factory? I know we do that elsewhere, but not sure with TypeInfo...

Reply:
I do like the idea of it as an inner interface `ChunkReader.Factory` and I agree, I wouldn't nest `TypeInfo` but rather make it `ChunkReader.TypeInfo` -- IMO this is clean and unambiguous.